### PR TITLE
fix 537

### PR DESCRIPTION
**Pull request by chicco785**
## Proposed changes

The function _is_ngsi_array was not computing correctly the data type casting in case the attribute type was not "Array" or "StructuredValue" (issue #537). issue was not surfacing cratedb, since crate is forcing cast to the column type anyhow.

also, the call was returning 200, despite the error given that "raise" was not issued. this last thing should be checked in better detail with @c0c0n3 since it may have been on purpose.

## Types of changes

What types of changes does your code introduce to the project: _Put
an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
      functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after
creating the PR. If you're unsure about any of them, don't hesitate to
ask. We're here to help! This is simply a reminder of what we are going
to look for before merging your code._

- [x] I have read the [CONTRIBUTING][contrib] doc
- [x] I have signed the [CLA][cla]
- [x] I have added tests that prove my fix is effective or that my
      feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in
      downstream modules

## Further comments

N/A

[cla]: https://raw.githubusercontent.com/orchestracities/ngsi-timeseries-api/master/individual-cla.pdf
    "Martel Open Source Software Individual Contributor License Agreement"
[contrib]: https://github.com/smartsdk/ngsi-timeseries-api/blob/master/CONTRIBUTING.md
    "Contributing to QuantumLeap"

### CODE REVIEW

File: src/translators/sql_translator.py [hunk 1]
```diff
@@ -397,6 +397,7 @@ def _insert_entity_rows(self, table_name: str, col_names: List[str],
             )
             self._insert_original_entities_in_failed_batch(
                 table_name, entities, e)
+            raise
```
Comment (chicco785) [inline src/translators/sql_translator.py:400]: introducing this, two translator tests fails:
https://app.circleci.com/pipelines/github/orchestracities/ngsi-timeseries-api/352/workflows/2116016d-0161-40ff-84d8-e6b17089c6d8/jobs/1410/tests#failed-test-1

https://app.circleci.com/pipelines/github/orchestracities/ngsi-timeseries-api/352/workflows/2116016d-0161-40ff-84d8-e6b17089c6d8/jobs/1410/tests#failed-test-3

Comment (chicco785) [reply src/translators/sql_translator.py:400]: for now let's comment it out, but let's talk about the logic here, and let's agree what should happen.

Comment (c0c0n3) [reply src/translators/sql_translator.py]: I don't think that `raise` statement should be there at all. should we zap it?

when the flow gets to that point, the outcome should be determined by the call to `_insert_original_entities_in_failed_batch`.

**Review by c0c0n3 — APPROVED** [review]

**Comment by c0c0n3** [comment]
> also, the call was returning 200, despite the error given that "raise" was not issued. this last thing should be checked in better detail with @c0c0n3 since it may have been on purpose.

That sounds fishy to me too. definitely not intentional in my opinion.

**Pull request merged**